barman/server.py

```diff
@@ -3044,34 +3046,67 @@ def get_backup_ext_info(self, backup_info):
-                else:
-                    backup_ext_info["next_backup_id"] = None
             except UnknownBackupIdException:
                 # no next_backup_id and previous_backup_id items
                 # means "Not available"
                 pass
             backup_ext_info.update(self.get_wal_info(backup_info))
+
+            backup_ext_info["retention_policy_status"] = None
             if self.enforce_retention_policies:
                 policy = self.config.retention_policy
                 backup_ext_info["retention_policy_status"] = policy.backup_status(
                     backup_info.backup_id
                 )
-            else:
-                backup_ext_info["retention_policy_status"] = None
-
             # Check any child timeline exists
             children_timelines = self.get_children_timelines(
                 backup_ext_info["timeline"], forked_after=backup_info.end_xlog
             )
 
             backup_ext_info["children_timelines"] = children_timelines
 
+            # If copy statistics are available, get summary
+            copy_stats = backup_info.copy_stats
+            if copy_stats:
+                copy_time = copy_stats.get("copy_time", 0)
+                if copy_time:
+                    backup_ext_info["copy_time"] = copy_time
+                    # Get analysis time if it is more than a second
+                    analysis_time = copy_stats.get("analysis_time", 0)
+                    if analysis_time is not None and analysis_time >= 1:
+                        backup_ext_info["analysis_time"] = analysis_time
+
+                    if backup_info.deduplicated_size is not None:
+                        estimated_throughput = backup_info.deduplicated_size / copy_time
+                        backup_ext_info["estimated_throughput"] = estimated_throughput
+                        number_of_workers = copy_stats.get("number_of_workers", 1)
+                        if number_of_workers > 1:
+                            backup_ext_info["number_of_workers"] = number_of_workers
+
+            # Info needed after incremental backups was released in Postgres 17
+            backup_ext_info["parent_backup_id"] = backup_info.parent_backup_id
+            backup_ext_info["children_backup_ids"] = backup_info.children_backup_ids
+            backup_ext_info["cluster_size"] = backup_info.cluster_size
+            backup_ext_info["mode"] = backup_info.mode
+
+            backup_chain = [backup for backup in backup_info.walk_to_root()]
+
+            chain_size = len(backup_chain)
+            # last is root
+            root_backup_info = backup_chain[-1]
+            backup_ext_info["root_backup_id"] = root_backup_info.backup_id
+            backup_ext_info["chain_size"] = chain_size
+            backup_ext_info["deduplication_ratio"] = backup_info.deduplication_ratio
+            backup_ext_info["est_dedup_size"] = (
+                backup_info.cluster_size * backup_info.deduplication_ratio
```

Comment:
To be really accurate here, we should do:

* `backup_info.size * backup_info.deduplication_ratio` for rsync-based;
* `backup_info.cluster_size * backup_info.deduplication_ratio` for postgres-based .

I mean, that's the math behind the "ratio" itself.

But after thinking more and more about this, I wonder if it's really worth to do that or not here. That's an estimation of resource savings.

I say that because I've been trying to think of an elegant way of "being more precise", but it's been hard to think of an elegant solution.

The best I could think of was something like:

* Turn `deduplication_ratio` property into `deduplication_info` property. It would return a dict with some keys instead of a single value:
   * `ratio`: the value that we already calculate right now
   * `total_size`: value of `size` variable of the existing property, which is either `size` or `cluster_size` depending on the method;
   * `est_dedup_size`: `ratio` * `total_size`.

But that seems a bit ugly to me.

Anyway, I think I'm just overthinking here.

Thoughts @andremagui @martinmarques ?

Reply:
I'm inclined to just leave it as it is, and add a comment above that line explaining why `backup_info.cluster_size` is being used.

As an example of what I mean, if you want to look at it, I took 2 rsync backups in a row and did not change anything in between the backups.

With the current math I get an output like this:

```console
$ barman show-backup support rsync_inc_4
Backup 20240731T212934:
  Backup Name            : rsync_inc_4
  Server Name            : support
  System Id              : 7382227563826631291
  Status                 : DONE
  PostgreSQL Version     : 170000
  PGDATA directory       : /var/lib/pgsql/17/data
  Estimated Cluster Size : 565.1 MiB

  Server information:
    Checksums            : off
    WAL summarizer       : on

  Tablespaces:
    custom_ts_1          : /var/lib/pgsql/17/tablespaces/custom_ts_1 (oid: 24678)
    custom_ts_2          : /var/lib/pgsql/17/tablespaces/custom_ts_2 (oid: 24686)

  Base backup information:
    Backup Method        : rsync-concurrent
    Backup Size          : 9.1 KiB (16.0 MiB with WALs)
    WAL Size             : 16.0 MiB
    Resource savings     : 565.1 MiB (100.00%)
```

If we replace `cluster_size` with `size` when method is rsync, so we follow the logic of deduplication ratio:

```console
$ barman show-backup support rsync_inc_4
Backup 20240731T212934:
  Backup Name            : rsync_inc_4
  Server Name            : support
  System Id              : 7382227563826631291
  Status                 : DONE
  PostgreSQL Version     : 170000
  PGDATA directory       : /var/lib/pgsql/17/data
  Estimated Cluster Size : 565.1 MiB

  Server information:
    Checksums            : off
    WAL summarizer       : on

  Tablespaces:
    custom_ts_1          : /var/lib/pgsql/17/tablespaces/custom_ts_1 (oid: 24678)
    custom_ts_2          : /var/lib/pgsql/17/tablespaces/custom_ts_2 (oid: 24686)

  Base backup information:
    Backup Method        : rsync-concurrent
    Backup Size          : 9.1 KiB (16.0 MiB with WALs)
    WAL Size             : 16.0 MiB
    Resource savings     : 565.3 MiB (100.00%)
```

I mean, it's a negligible difference in "Resource savings"

Reply:
Yeah, here was a little hard to be accurate for both! I mean, i thought about getting the deduplication_ratio for both, using the correct sizes for each one! But calculating both using cluster_size for an estimation of the resource savings. By doing like this, we would not get any awkward results like in the last example above.
Estimated Cluster Size : 565.1 MiB
Resource savings     : 565.3 MiB (100.00%)

That was the only reason why i only created the ratio property! But yeah, i really don't know a best option here! All have little "flaws".